src/registry/live_check.rs

```diff
@@ -1,112 +1,340 @@
+        DiagnosticMessages::from(Error::OutputError {
+            error: format!(
+                "Failed to load `defaults/live_check_templates/weaver.yaml`: {}",
+                e
+            ),
+        })
+    })?;
+    let engine = TemplateEngine::new(config, loader, Params::default());
+
+    // Prepare the ingester
+    let mut stream_mode = false;
+    let ingester = match args.ingester {
+        IngesterType::AttributeFile => {
+            let path = match &args.input {
+                Some(p) => Ok(p),
+                None => Err(Error::IngestError {
+                    error: "No input path provided".to_owned(),
+                }),
+            }?;
+            AttributeFileIngester::new(path).ingest(logger.clone())?
+        }
+
+        IngesterType::AttributeStdin => {
+            if args.stream.is_none() {
+                stream_mode = true;
             }
-            OtlpRequest::Traces(_traces) => {
-                // ToDo Implement the checking logic for traces
-                println!("Trace Request received");
+            AttributeStdinIngester::new().ingest(logger.clone())?
+        }
+
+        IngesterType::AttributeJsonFile => {
+            let path = match &args.input {
+                Some(p) => Ok(p),
+                None => Err(Error::IngestError {
+                    error: "No input path provided".to_owned(),
+                }),
+            }?;
+            AttributeJsonFileIngester::new(path).ingest(logger.clone())?
+        }
+
+        IngesterType::AttributeJsonStdin => {
+            if args.stream.is_none() {
+                stream_mode = true;
             }
-            OtlpRequest::Stop(reason) => {
-                logger.warn(&format!("Stopping the listener, reason: {}", reason));
-                // ToDo Generate the report here
-                break;
+            AttributeJsonStdinIngester::new().ingest(logger.clone())?
+        }
+
+        IngesterType::AttributeOtlp => {
+            if args.stream.is_none() {
+                stream_mode = true;
             }
-            OtlpRequest::Error(error) => {
-                diag_msgs.extend(DiagnosticMessages::from_error(error));
-                break;
+            (AttributeOtlpIngester {
+                otlp_grpc_address: args.otlp_grpc_address.clone(),
+                otlp_grpc_port: args.otlp_grpc_port,
+                admin_port: args.admin_port,
+                inactivity_timeout: args.inactivity_timeout,
+            })
+            .ingest(logger.clone())?
+        }
+
+        IngesterType::GroupFile => {
+            return Err(DiagnosticMessages::from(Error::OutputError {
+                error: "Invalid ingester type".to_owned(),
+            }))
+        }
+    };
+
+    // If this is a stream - process the attributes one by one
+    // File output is not supported in stream mode
+    if let OutputDirective::File = output_directive {
+        stream_mode = false;
+    }
+
+    if stream_mode {
```

Comment:
I wonder if we can "hide" stream mode behind a trait that Ingestor would implement...

Thoughts for future refactor